internal/provider/group_resource_test.go

```diff
@@ -0,0 +1,277 @@
+package provider
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseHttpResponse(t *testing.T) {
+	t.Run("Basic Test", func(t *testing.T) {
+		g := GroupResourceModel{
+			Name: types.StringValue("group1"),
+			URL:  types.StringValue("/api/v2/groups/24/"),
+		}
+		body := []byte(`{"name": "group1", "url": "/api/v2/groups/24/", "description": ""}`)
+		err := g.ParseHttpResponse(body)
+		assert.NoError(t, err)
```

Comment:
I think we probably also want to test the values of the `GroupResourceModel` here to be sure they have been updated correctly. You can start with it empty to ensure all values are added (e.g., `g := GroupResourceModel{}`).